main/app.js

```diff
@@ -1,6 +1,69 @@
-String.prototype.hasVowels = function() {};
-String.prototype.toUpper = function() {};
-String.prototype.toLower = function() {};
-String.prototype.ucFirst = function() {};
+String.prototype.hasVowels = function() {
+    return /[aeiou]/.test(this);
+};
+String.prototype.toUpper = function() {
+    var converted = '';
+    for (var index = 0; index < this.length; index++) {
+        converted += String.fromCharCode(this.charCodeAt(index) & 223);
+    }
+    return converted;
+};
+String.prototype.toLower = function() {
+    var converted = '';
+    for (var index = 0; index < this.length; index++) {
+        converted += String.fromCharCode(this.charCodeAt(index) | 32);
+    }
+    return converted;
+};
+String.prototype.ucFirst = function() {
+    return /[a-zA-Z]/.exec(this)[0] + this.substring(1).toLower();
```

Comment:
Wow! Please add comments here and split the implementation over several lines. This your one-liner; I hail oooo!!